### ISSUE TITLE
Compliance upload cli does not check the version of a profile being uploaded

### ISSUE THREAD

**Issue by srb3** [inspec/inspec#3246]
### 🎛 Description

bug report

Submitting on behalf of a customer:

when using chefdk-3.1.0-1.el7.x86_64 uploading a new version of an inspec profile that is already uploaded fails stating it already exists.

### Expected behaviour:
New version of profile is uploaded
Actual behaviour:
Upload fails, newer version of the profile is not uploaded (--overwrite is suggested)
Output:

 ```
$ inspec compliance profiles
== Available profiles:
* Standard build compliance v1.8.0 (chefbot/inspec-demo-profile)

$ vi inspec.yml
$ grep -E "title|version" inspec.yml
title: Standard build compliance 1.9.0 <-Changed title
version: 1.9.0 <-- version bumped

$ inspec compliance upload ./inspec-demo-profile
Profile is already vendored. Use --overwrite.
I, [2018-07-24T11:09:42.081881 #1588]  INFO -- : Checking profile in ./inspec-demo-profile
I, [2018-07-24T11:09:42.082157 #1588]  INFO -- : Metadata OK.
I, [2018-07-24T11:09:42.086905 #1588]  INFO -- : Found 1 controls.
W, [2018-07-24T11:09:42.086957 #1588]  WARN -- : Control break the rules has no description
Profile is valid
Profile exists on the server, use --overwrite
Found 1 error(s)

$ inspec compliance profiles
== Available profiles:
* Standard build compliance v1.8.0 (chefbot/inspec-demo-profile)
```
### 🌍 InSpec and Platform Version
inspec version: 2.1.72
RHEL 7

### 🤔 Replication Case

Create a profile and upload it to an automate server using the inspec cli.
Make changes to the profile and increase version number.
Retry profile upload.

### 💁 Possible Solutions

adding the following to lib/bundles/inspec-compliance/cli.rb in the upload method
seems to solve the issue

```
     # read profile version from inspec.yml
     profile_version = profile.params[:version]

      # check that the profile is not uploaded already,
      # confirm upload to the user (overwrite with --force)
      if Compliance::API.exist?(config, "#{config['owner']}/#{profile_name}##{profile_version}") && !options['overwrite']
        error.call('Profile exists on the server, use --overwrite')
      end

```
these additions allow the `exist?` method in `Compilance::API` to evaluate on the version of the profile being uploaded
```
    # verifies that a profile
    def self.exist?(config, profile)
      owner, id, ver = profile_split(profile)

      # ensure that we do not manipulate the configuration object
      user_config = config.dup
      user_config['owner'] = owner
      _msg, profiles = Compliance::API.profiles(user_config)

      if !profiles.empty?
        profiles.any? do |p|
          profile_owner = p['owner_id'] || p['owner']
          profile_owner == owner &&
            p['name'] == id &&
            (ver.nil? || p['version'] == ver)
        end
      else
        false
      end
    end

```

**Comment by clintoncwolfe** [comment]
FYI duplicates #3137

**Comment by clintoncwolfe** [comment]
Fixed by #3252
